include/zephyr/drivers/pwm.h

```diff
@@ -263,15 +455,36 @@ static inline int pwm_pin_set_usec(const struct device *dev, uint32_t pwm,
 		return -ENOTSUP;
 	}
 
-	return pwm_pin_set_cycles(dev, pwm, (uint32_t)period_cycles,
-				  (uint32_t)pulse_cycles, flags);
+	return pwm_set_cycles(dev, channel, (uint32_t)period_cycles,
+			      (uint32_t)pulse_cycles, flags);
+}
+
+/**
+ * @brief Set the period and pulse width in microseconds from a struct
+ *        pwm_dt_spec.
+ *
+ * This is equivalent to:
+ *
+ *     pwm_set_usec(spec->dev, spec->channel, spec->period / NSEC_PER_USEC,
+ *                  pulse, spec->flags)
+ *
+ * @param[in] spec PWM specification from devicetree.
+ * @param pulse Pulse width (in microseconds) set to the PWM.
+ *
+ * @return A value from pwm_set_usec().
+ */
+static inline int pwm_set_usec_dt(const struct pwm_dt_spec *spec,
+				  uint32_t pulse)
```

Comment:
I was wondering if `*_dt` functions that allow setting also `period` could be provided, too (for example, we could have `pwm_set_usec_pulse_dt` for setting pulse only and `pwm_set_usec_dt` for setting both). Those could be useful in cases   like blinky_pwm or when PWM drives a buzzer.

Reply:
updated PR, please review